core/thegraph/state.go

```diff
@@ -48,8 +48,16 @@ type (
 		SocketUpdates []SocketUpdates `graphql:"socketUpdates(first: 1, orderBy: blockNumber, orderDirection: desc)"`
 	}
 
+	// The indexed operator state consists of both mutable properties (socket) and  immutable properties
+	// (everyhing else: pubkeyG1, pubkeyG2, id). For the socket, we always want the latest value, irrespective
+	// of the reference block number. For the immutable properties, we can also use the value from the latest block
+	// since value cannot change. Thus, we always pull the state from the latest block indexed by the subgraph.
+	//
+	// Note that the deregistrationBlockNumber will only be set if the operator has deregistered from all quorums. By using
+	// the latest block, we allow the false-positive case where an operator was deregistered from all quorums at the reference
+	// block, but then re-registered afterward. We filter out these operators in GetIndexedOperatorState.
```

Comment:
Can this call out that it may over-fetch (i.e. a superset of operators that are alive at reference block), but never under-fetch?

Reply:
Done